code/forms/JsonField.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * Class JsonField combines form inputs into a key-value pair
+ */
+class JsonField extends CompositeField
+{
+    public function __construct($name, $children = null)
+    {
+        $this->setName($name);
+
+        if ($children instanceof FieldList) {
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
+        } elseif (is_array($children)) {
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
```

Comment:
This logic is the same as the previous if statement - can you just combine? `if (is_array($children) || $children instanceof FieldList) {`